esmvalcore/cmor/_fixes/icon/icon.py

```diff
@@ -0,0 +1,163 @@
+        """Fix height coordinate of cube."""
+        air_pressure_points = plev_points_cube.core_data()
+
+        # Get bounds from half levels and reshape array
+        air_pressure_bounds = plev_bounds_cube.core_data()
+        air_pressure_bounds = da.stack(
+            (air_pressure_bounds[:, :-1], air_pressure_bounds[:, 1:]), axis=-1)
+
+        # Setup air pressure coordinate with correct metadata and add to cube
+        air_pressure_coord = iris.coords.AuxCoord(
+            air_pressure_points,
+            bounds=air_pressure_bounds,
+            var_name='plev',
+            standard_name='air_pressure',
+            long_name='pressure',
+            units=plev_points_cube.units,
+            attributes={'positive': 'down'},
+        )
+        cube.add_aux_coord(air_pressure_coord, np.arange(cube.ndim))
+
+        # Add 'positive' attribute to height so iris can determine Z axis
+        # correctly
+        cube.coord('height').attributes['positive'] = 'down'
+
+        # Reverse height axis so that index 0 is surface level
+        cube = iris.util.reverse(cube, 'height')
+        return cube
+
+    @staticmethod
+    def _fix_lat_lon(cube, lat_name, lon_name):
+        """Fix latitude and longitude coordinates of cube."""
+        lat = cube.coord(lat_name)
+        lon = cube.coord(lon_name)
+
+        # Fix metadata
+        lat.var_name = "lat"
+        lon.var_name = "lon"
+        lat.standard_name = "latitude"
+        lon.standard_name = "longitude"
+        lat.long_name = "latitude"
+        lon.long_name = "longitude"
+        lat.convert_units('degrees')
+        lon.convert_units('degrees')
+        lat.units = "degrees_north"
+        lon.units = "degrees_east"
+
+        # If grid is not unstructured, no further changes are necessary
+        if cube.coord_dims(lat) != cube.coord_dims(lon):
+            return
+        horizontal_coord_dims = cube.coord_dims(lat)
+        if len(horizontal_coord_dims) != 1:
+            return
+
+        # Add dimension name for cell index used to store the unstructured grid
+        index_coord = iris.coords.DimCoord(
+            np.arange(cube.shape[horizontal_coord_dims[0]]),
+            var_name="i",
+            long_name=("first spatial index for variables stored on an "
+                       "unstructured grid"),
+            units="1",
+        )
+        cube.add_dim_coord(index_coord, horizontal_coord_dims)
+
+    @staticmethod
+    def _fix_time(cube):
+        """Fix time coordinate of cube."""
+        t_coord = cube.coord("time")
+        t_coord.var_name = 'time'
+        t_coord.standard_name = 'time'
+        t_coord.long_name = 'time'
+
+        # Convert units to CF format
+        t_unit = t_coord.attributes.pop("invalid_units")
+        timestep, _, t_fmt_str = t_unit.split(" ")
+        new_t_unit_str = f"{timestep} since 1850-01-01"
+        new_t_unit = cf_units.Unit(new_t_unit_str, calendar="standard")
```

Comment:
It's actually `proleptic_gregorian`, so we probably should use this. There is no way of reading this from the file, since iris removes the information and just saves the `invalid_units` without the calender, e.g.

```
DimCoord(array([20110201.]), standard_name='time', units=Unit('unknown'), var_name='time', attributes={'invalid_units': 'day as %Y%m%d.%f'})
```

Reply:
Ah yes, I'd consider this an iris bug. Perhaps raise it over there when you come back from vacation.

Reply:
Oh but definitely put the proleptic_gregorian. That really could have an impact for spinup runs.